src/prefect/cli/deployment.py

```diff
@@ -93,37 +93,71 @@ async def get_deployment(client: OrionClient, name, deployment_id):
+                    )
                     if res.concurrency_limit != work_queue_concurrency:
-                        app.console.print(
-                            f"Work queue {work_queue_name!r} already exists with a concurrency limit of {res.concurrency_limit}, this limit is being updated...",
-                            style="red",
-                        )
+                        if work_pool_name is None:
+                            app.console.print(
+                                f"Work queue {work_queue_name!r} already exists with a concurrency limit of {res.concurrency_limit}, this limit is being updated...",
+                                style="red",
+                            )
+                        else:
+                            app.console.print(
+                                f"Work queue {work_queue_name!r} in work pool {work_pool_name!r} already exists with a concurrency limit of {res.concurrency_limit}, this limit is being updated...",
+                                style="red",
+                            )
                 await client.update_work_queue(
                     res.id, concurrency_limit=work_queue_concurrency
                 )
-                app.console.print(
-                    f"Updated concurrency limit on work queue {work_queue_name!r} to {work_queue_concurrency}",
-                    style="green",
-                )
+                if work_pool_name is None:
+                    app.console.print(
+                        f"Updated concurrency limit on work queue {work_queue_name!r} to {work_queue_concurrency}",
+                        style="green",
+                    )
+                else:
+                    app.console.print(
+                        f"Updated concurrency limit on work queue {work_queue_name!r} in work pool {work_pool_name!r} to {work_queue_concurrency}",
+                        style="green",
+                    )
             except Exception as exc:
                 exit_with_error(
-                    f"Failed to set concurrency limit on work queue {work_queue_name}."
+                    f"Failed to set concurrency limit on work queue {work_queue_name!r} in work pool {work_pool_name!r}."
                 )
         elif work_queue_concurrency:
             app.console.print(
                 f"No work queue set! The concurrency limit cannot be updated."
             )
 
 
+async def check_work_pool_exists(work_pool_name: Optional[str]):
+    if work_pool_name is not None and PREFECT_EXPERIMENTAL_ENABLE_WORK_POOLS.value():
+        async with get_client() as client:
+            try:
+                await client.read_work_pool(work_pool_name=work_pool_name)
+            except ObjectNotFound:
+                exit_with_error(
+                    f"This deployment specifies a work pool name of {work_pool_name!r}, but no such "
+                    "work pool exists. To create a work pool, visit the UI and create "
```

Comment:
don't we also have a CLI for this that we can display the command for?

Reply:
Good call! Added in [aeaa34b](https://github.com/PrefectHQ/prefect/pull/8413/commits/aeaa34b83eecae6b45a4af69b5b30d1f46bfe131)